errors.go

```diff
@@ -153,7 +153,7 @@ func GetSystemErrorCode(err error) SystemErrCode {
 		return se.Code()
 	}
 
-	return ErrCodeUnexpected
+	return ErrCodeInvalid
```

Comment:
I think this line should remain the same, and instead you should have an if at the very top,
```go
if err == nil {
  return ErrCodeInvalid
}
```